apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryDataReader.cs

```diff
@@ -14,197 +14,207 @@
 
         /// <inheritdoc />
-        public override char GetChar(int ordinal)
-        {
-            throw new NotImplementedException();
-        }
+        public override int RecordsAffected => -1;
 
         /// <inheritdoc />
-        public override long GetChars(int ordinal, long dataOffset, char[] buffer, int bufferOffset, int length)
-        {
-            throw new NotImplementedException();
-        }
+        public override bool GetBoolean(int i) => GetFieldValue<bool>(i);
 
         /// <inheritdoc />
-        public override Guid GetGuid(int ordinal)
-        {
-            throw new NotImplementedException();
-        }
+        public override byte GetByte(int i) => GetFieldValue<byte>(i);
 
         /// <inheritdoc />
-        public override short GetInt16(int ordinal)
-        {
-            throw new NotImplementedException();
-        }
+        public override long GetBytes(int i, long fieldOffset, byte[] buffer, int bufferoffset, int length) => throw
+            new NotSupportedException("BigQuery does not support conversion to byte arrays.");
 
         /// <inheritdoc />
-        public override int GetInt32(int ordinal)
-        {
-            throw new NotImplementedException();
-        }
+        public override char GetChar(int i) => GetFieldValue<char>(i);
 
         /// <inheritdoc />
-        public override long GetInt64(int ordinal)
-        {
-            throw new NotImplementedException();
-        }
+        public override long GetChars(int i, long fieldoffset, char[] buffer, int bufferoffset, int length) => throw
+            new NotSupportedException("BigQuery does not support conversion to char arrays.");
 
         /// <inheritdoc />
-        public override DateTime GetDateTime(int ordinal)
-        {
-            throw new NotImplementedException();
-        }
+        public override string GetDataTypeName(int i) => Current?.Schema.Fields[i].GetFieldType().ToString();
 
         /// <inheritdoc />
-        public override string GetString(int ordinal)
-        {
-            throw new NotImplementedException();
-        }
+        public override DateTime GetDateTime(int i) => GetFieldValue<DateTime>(i);
 
         /// <inheritdoc />
-        public override decimal GetDecimal(int ordinal)
-        {
-            throw new NotImplementedException();
-        }
+        public override decimal GetDecimal(int i) => GetFieldValue<decimal>(i);
 
         /// <inheritdoc />
-        public override double GetDouble(int ordinal)
-        {
-            throw new NotImplementedException();
-        }
+        public override double GetDouble(int i) => GetFieldValue<double>(i);
 
         /// <inheritdoc />
-        public override float GetFloat(int ordinal)
+        public override IEnumerator GetEnumerator()
         {
-            throw new NotImplementedException();
+#if NET45 || NET451
+            return new DbEnumerator(this);
```

Comment:
Do I need to look into what DbEnumerator does? Does it ever start iterating from the start again?

Reply:
I dont *think* so.  It seems to be rarely used externally, but I remember seeing places where its used internally.  Is there a concern with re-enumerating if it does that?